### DISCUSSION TITLE
[IDEA] Menu-Bar Item Drop-Down near right display-border | possible Solution

### DISCUSSION THREAD

**Discussion by toraxmalu** [TiddlyWiki/TiddlyWiki5#5533] [Ideas]
While playing with the menu-bar-plugin I noticed an annoying behavior: Pop-ups are cut-off, when the menu-bar-item is near the right window border. This can be e.g. with style "float: right;". See here [Google-Groups](https://groups.google.com/g/tiddlywiki/c/fVRZkl9bqBU).

While investigating I noticed the behavior of the RevealWidget. Maybe a solution / fix could be with the if-statement added to the default-section in the function RevealWidget.prototype.positionPopup in reveal.js:

```
default: // Below
	left = this.popup.left;
	top = this.popup.top + this.popup.height;

	if ((this.popup.left + domNode.offsetWidth) > window.innerWidth) {
		left = this.popup.left + this.popup.width - domNode.offsetWidth;
	}
	break;
```

The if-clause checks, weather the right frame is passing the display-area. In that case it replaces the left-value with a new value, that makes the pop-up right-aligned. Later in the code the widget checks, weather it is rendered in negative x-value and replacing the left with the 0, if that isn't allowed explicit (positionAllowNegative).

What are your ideas?

Cheers ToraxMalu

**Comment by Jermolene** [comment]
Hi @toraxmalu I've made a slightly different fix so that one can add a "dropdown-position" field set to "belowleft" to the custom menu item.

**Reply by Jermolene** [reply]
The commit is here: https://github.com/Jermolene/TiddlyWiki5/commit/81546c5bf49c6644e4e69edc433208d5ba238743

**Reply by toraxmalu** [reply]
Interesting solution. You "simply" add a new field to pass the position-parameter. Thanks and cheers.
